base_template.cpp

```diff
@@ -9,6 +9,10 @@ using namespace std;
 
 typedef long long ll;
 
+const ll N = 1e6;
+const ll INF = 1e9;
```

Comment:
Maybe we should set `INF = 1e9 + 1`, so if a problem limits a variable to 1e9 (as a lot of problems do), we can distinguish between a variable with that maximum value and INF.

Actually, it would be better to write
```suggestion
const ll INF = ll(1e9) + 1;
```
to make sure we won't generate a bug if we change INF to 1e18 + 1 in a problem (`1e18 + 1 == 1e18` because of floating point precision).

Reply:
Actually, what if we use 64 bits always? I don't think it will give use any problem. And to avoid precision issues, I think it would be better to define it as
`const ll INF = 1LL << 62;`

Reply:
I think this is even better! After changing that, the template will be good enough for now.

Reply:
I ended up using `1LL << 61` to avoid overflow when doing `INF + INF`.